File: ruby/bob.rb
class Bob

  def self.hey(remark)
    return "Sure." if BobBrain.new(remark).question?
    return "Whoa, chill out!" if BobBrain.new(remark).shouting?
    return "Fine. Be that way!" if BobBrain.new(remark).silence?
    return "Whatever."
  end


end

class BobBrain
  def initialize(remark)
    @remark=remark.split("")
  end

  def shouting?
    !(@remark.any? {|letter| ("a".."z").include?(letter)}) && @remark.any? {|letter| ("A".."Z").include?(letter)}
  end

  def question?
    @remark.last =="?" && !(shouting?)
  end

  def silence?
    @remark.join("").strip.empty?
  end

end

File: js/bob.js
Bob = function() {};

Bob.prototype.hey = function(statement) {
  if (statement.trim()=="") {
    return "Fine. Be that way!"
  }
  else if (statement.toUpperCase() == statement && statement.toLowerCase() != statement) {
    return "Whoa, chill out!"
  }
  else if (statement.charAt(statement.length-1)=="?") {
    return "Sure."
  }
  else {
    return "Whatever."
  };
};

module.exports = Bob
